src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,31 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+	public String STRING1 = "123456789";
+    public String STRING2 = "Person phone numbers should only contain numbers";
+    public String STRING3 = "\\d+";
+
+    public String value;
+    private boolean isPrivate;
+    //constructor
+    public Contact(String STRING1, String STRING2, String STRING3, String value, boolean isPrivate){
+    	this.STRING1 = STRING1;
```

Comment:
Based on the values you are passing in the subclasses STRING2 and STRING3 seem to contain the validation regex and the error message. You must use them to validate the address. If not the superclass is not complete. Consider validating the values in the constructor.